src/actions.rs

```diff
@@ -0,0 +1,497 @@
+            PointerOrigin::Viewport => WDActions::PointerOrigin::Viewport,
+            PointerOrigin::Pointer => WDActions::PointerOrigin::Pointer,
+            PointerOrigin::WebElement(e) => WDActions::PointerOrigin::Element(
+                webdriver::common::WebElement(e.element_id().into()),
+            ),
+        }
+    }
+}
+
+impl PointerAction {
+    fn into_item(self) -> WDActions::PointerActionItem {
+        match self {
+            PointerAction::Pause { duration } => WDActions::PointerActionItem::General(
+                WDActions::GeneralAction::Pause(WDActions::PauseAction {
+                    duration: Some(duration.as_millis() as u64),
+                }),
+            ),
+            PointerAction::Down { button } => WDActions::PointerActionItem::Pointer(
+                WDActions::PointerAction::Down(WDActions::PointerDownAction { button }),
+            ),
+            PointerAction::Up { button } => WDActions::PointerActionItem::Pointer(
+                WDActions::PointerAction::Up(WDActions::PointerUpAction { button }),
+            ),
+            PointerAction::Move {
+                duration,
+                origin,
+                x,
+                y,
+            } => WDActions::PointerActionItem::Pointer(WDActions::PointerAction::Move(
+                WDActions::PointerMoveAction {
+                    duration: duration.map(|x| x.as_millis() as u64),
+                    origin: origin.into_wd_pointerorigin(),
+                    x: Some(x),
+                    y: Some(y),
+                },
+            )),
+            PointerAction::Cancel => {
+                WDActions::PointerActionItem::Pointer(WDActions::PointerAction::Cancel)
+            }
+        }
+    }
+}
+
+/// A sequence containing `Null` actions.
+#[derive(Debug, Clone)]
+pub struct NullActions {
+    /// An identifier to distinguish this sequence from others.
+    ///
+    /// Choose a meaningful string as it may be useful for debugging.
+    id: String,
+    /// The list of actions for this sequence.
+    actions: Vec<NullAction>,
+}
+
+impl NullActions {
+    /// Create a new NullActions sequence.
+    pub fn new(id: &str) -> Self {
+        Self {
+            id: id.to_string(),
+            actions: Vec::new(),
+        }
+    }
+}
+
+impl From<NullActions> for ActionSequence {
+    fn from(na: NullActions) -> Self {
+        ActionSequence(WDActions::ActionSequence {
+            id: na.id,
+            actions: WDActions::ActionsType::Null {
+                actions: na.actions.into_iter().map(|x| x.into_item()).collect(),
+            },
+        })
+    }
+}
+
+/// A sequence containing `Key` actions.
+#[derive(Debug, Clone)]
+pub struct KeyActions {
+    /// An identifier to distinguish this sequence from others.
+    ///
+    /// Choose a meaningful string as it may be useful for debugging.
+    id: String,
+    /// The list of actions for this sequence.
+    actions: Vec<KeyAction>,
+}
+
+impl KeyActions {
```

Comment:
Should we also implement `Default` for this and the other `*Actions` types with some default identifier, or is that unlikely to be useful/meaningful?

Reply:
If they don't need to be unique, then that's probably fine. I'll do some testing to find out.

Reply:
Ok so they do need to be unique. In most cases you'd only have one of each device type but it's still possible to have multiple and in that case the id needs to be unique to the device.

I think we need the caller to specify the id.

Reply:
Ah, good catch! In that case, let's also document that requirement.